tagid.go

```diff
@@ -4,105 +4,125 @@
 package swid
 
 import (
-	"encoding/hex"
 	"encoding/json"
 	"encoding/xml"
 	"errors"
 	"fmt"
 
 	"github.com/fxamacker/cbor/v2"
+	"github.com/google/uuid"
 )
 
-// TagID is the type of a tag identifier. Allowed formats (enforced via
-// checkTagID) are string or [16]byte
+// TagID is the type of a tag identifier. Allowed formats are string or
+// a valid universally unique identifier (UUID) as defined by RFC4122.
 type TagID struct {
 	val interface{}
 }
 
-// NewTagID returns a TagID initialized with the supplied value v
-// v is either a string or a [16]byte
 func NewTagID(v interface{}) *TagID {
-	if checkTagID(v) != nil {
+	switch t := v.(type) {
+	case string:
```

Comment:
See my comment on Line 36, I think, what you have done here on line 24 is correct and as per the draft...

Reply:
not sure I understand this comment: L24 ends up calling L36...

Reply:
Agree, passed string is first interpreted as UUID and if not successful, treated as a plane string, so no issues here!